cmd/juju/crossmodel/show_test.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package crossmodel_test
+
+import (
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/cmd/juju/crossmodel"
+	"github.com/juju/juju/testing"
+)
+
+type showSuite struct {
+	BaseCrossModelSuite
+	mockAPI *mockShowAPI
+}
+
+var _ = gc.Suite(&showSuite{})
+
+func (s *showSuite) SetUpTest(c *gc.C) {
+	s.BaseCrossModelSuite.SetUpTest(c)
+
+	s.mockAPI = &mockShowAPI{
+		serviceTag: "service-hosted-db2",
+		desc:       "IBM DB2 Express Server Edition is an entry level database system",
+	}
+}
+
+func (s *showSuite) runShow(c *gc.C, args ...string) (*cmd.Context, error) {
+	return testing.RunCommand(c, crossmodel.NewShowEndpointsCommandForTest(s.mockAPI), args...)
+}
+
+func (s *showSuite) TestShowNoUrl(c *gc.C) {
+	s.assertShowError(c, nil, ".*must specify endpoint URL.*")
+}
+
+func (s *showSuite) TestShowApiError(c *gc.C) {
+	s.mockAPI.msg = "fail"
+	s.assertShowError(c, []string{"local:/u/fred/prod/db2"}, ".*fail.*")
+}
+
+func (s *showSuite) TestShowConversionError(c *gc.C) {
+	s.mockAPI.serviceTag = "invalid_tag"
+	s.assertShowError(c, []string{"local:/u/fred/prod/db2"}, ".*could not parse service tag.*")
+}
+
+func (s *showSuite) TestShowYaml(c *gc.C) {
+	s.assertShow(
+		c,
+		[]string{"local:/u/fred/prod/db2", "--format", "yaml"},
+		`
+hosted-db2:
+  endpoints:
+    db2:
+      interface: hhtp
+      role: role
+    log:
+      interface: http
+      role: role
+  description: IBM DB2 Express Server Edition is an entry level database system
+`[1:],
+	)
+}
+
+func (s *showSuite) TestShowTabular(c *gc.C) {
+	s.assertShow(
+		c,
+		[]string{"local:/u/fred/prod/db2", "--format", "tabular"},
+		`
+SERVICE     DESCRIPTION                     ENDPOINT  INTERFACE  ROLE
+hosted-db2  IBM DB2 Express Server Edition  db2       hhtp       role
```

Comment:
I don't think we should duplicate the non-wrapped columns. looks a bit like there's two items, but there's really just one spanning two rows